tests/websockets/subproto.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { test, expect } from '../../playwright';
+import { buildWebsocketCommonLocators } from '../utils/page/locators';
+import { waitForPredicate } from '../utils/wait';
+
+const BRU_REQ_NAME = /^ws-test-request-with-subproto$/;
+
+test.describe.serial('subprotocol tests', () => {
+  test('Only connect if a valid subprotocol is sent with the request', async ({ pageWithUserData: page, restartApp }) => {
+    const locators = buildWebsocketCommonLocators(page);
+    const clearText = async (text: string) => {
+      for (let i = text.length; i > 0; i--) {
+        await page.keyboard.press('Backspace');
+      }
+    };
+
+    const originalProtocol = 'soap';
+    const wrongProtocol = 'wap';
+
+    // Open the needed request and keep the headers tab in focus for modifications
+    await page.locator('#sidebar-collection-name').click();
+    await page.getByTitle(BRU_REQ_NAME).click();
+    await page.getByRole('tab', { name: 'Headers1' }).click();
+
+    // Check if the original / correct protocol is in place and then send a request
+    await expect(page.locator('pre').filter({ hasText: originalProtocol })).toBeAttached();
+    await locators.runner().click();
+
+    // Check the messages to confirm we ended up connecting
+    await waitForPredicate(() => locators.messages().count().then((count) => count > 0));
+    const messages = await locators.messages().all();
+    expect(await messages[0].locator('.text-ellipsis').innerText()).toMatch(/^(Connected to)/);
```

Comment:
I think we should just do this playwright way of handling dynamic UI tests.

This way, Playwright will:
- Wait for the element to appear.
- Retry until the text matches or timeout expires.

```suggestion
    const message = await locators.messages().first(); // or .nth(0)
    await expect(message.locator('.text-ellipsis')).toHaveText(/^(Connected to)/);
```

Reply:
It works, I hope it isn't flaky like the original `locators.messages()`